### DISCUSSION TITLE
Maintenance -> fix tree results in internal server error since several versions now. (minor bug in my understanding)

### DISCUSSION THREAD

**Discussion by axeljerabek** [LycheeOrg/Lychee#2496] [Bugs] [answered]
### Lychee version

v5.5.1

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

Internal Server Error
Illuminate\Database\Eloquent\MissingAttributeException
The attribute [cover_id] either does not exist or was not retrieved for model [App\Models\BaseAlbumImpl].
POST media.jerabek.fi
PHP 8.3.6 — Laravel 11.14.0

Collapse
vendor frames

Illuminate\Database\Eloquent\Concerns\HasAttributes
:503
throwMissingAttributeExceptionIfApplicable
 
Illuminate\Database\Eloquent\Concerns\HasAttributes
:483
getAttribute
2 vendor frames collapsed

App\Models\Extensions\ForwardsToParentImplementation
:348
getAttribute
 
Illuminate\Database\Eloquent\Relations\Relation
:285
array_map
 
Illuminate\Support\Arr
:605
map
 
Illuminate\Support\Collection
:759
map
 
Illuminate\Database\Eloquent\Relations\Relation
:284
getKeys
 
Illuminate\Database\Eloquent\Relations\HasOneOrMany
:105
addEagerConstraints
 
Illuminate\Database\Eloquent\Builder
:787
eagerLoadRelation
 
Illuminate\Database\Eloquent\Builder
:765
eagerLoadRelations
 
Illuminate\Database\Eloquent\Builder
:731
get
 
Kalnoy\Nestedset\QueryBuilder
:873
fixTree
9 vendor frames collapsed

App\Livewire\Components\Modules\Maintenance\FixTree
:49
do
 
Illuminate\Container\BoundMethod
:36
Illuminate\Container\{closure}
 
Illuminate\Container\Util
:41
unwrapIfClosure
 
Illuminate\Container\BoundMethod
:93
callBoundMethod
 
Illuminate\Container\BoundMethod
:35
call
 
Livewire\Wrapped
:23
__call
 
Livewire\Mechanisms\HandleComponents\HandleComponents
:474
callMethods
 
Livewire\Mechanisms\HandleComponents\HandleComponents
:101
update
 
Livewire\LivewireManager
:97
update
 
Livewire\Mechanisms\HandleRequests\HandleRequests
:94
handleUpdate
 
Illuminate\Routing\ControllerDispatcher
:46
dispatch
 
Illuminate\Routing\Route
:260
runController
 
Illuminate\Routing\Route
:206
run
 
Illuminate\Routing\Router
:806
Illuminate\Routing\{closure}
 
Illuminate\Pipeline\Pipeline
:144
Illuminate\Pipeline\{closure}
14 vendor frames collapsed

App\Http\Middleware\QueryStringFixer
:40
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
1 vendor frame collapsed

App\Http\Middleware\DisableCSP
:57
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Routing\Middleware\SubstituteBindings
:50
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Foundation\Http\Middleware\VerifyCsrfToken
:88
handle
4 vendor frames collapsed

App\Http\Middleware\VerifyCsrfToken
:44
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Session\Middleware\AuthenticateSession
:67
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\View\Middleware\ShareErrorsFromSession
:49
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Session\Middleware\StartSession
:121
handleStatefulRequest
 
Illuminate\Session\Middleware\StartSession
:64
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse
:37
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Cookie\Middleware\EncryptCookies
:75
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
12 vendor frames collapsed

App\Http\Middleware\AcceptContentType
:75
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Pipeline\Pipeline
:119
then
 
Illuminate\Routing\Router
:805
runRouteWithinStack
 
Illuminate\Routing\Router
:784
runRoute
 
Illuminate\Routing\Router
:748
dispatchToRoute
 
Illuminate\Routing\Router
:737
dispatch
 
Illuminate\Foundation\Http\Kernel
:200
Illuminate\Foundation\Http\{closure}
 
Illuminate\Pipeline\Pipeline
:144
Illuminate\Pipeline\{closure}
 
Livewire\Features\SupportDisablingBackButtonCache\DisableBackButtonCacheMiddleware
:19
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Bepsvpt\SecureHeaders\SecureHeadersMiddleware
:18
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull
:27
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Foundation\Http\Middleware\TrimStrings
:47
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Http\Middleware\ValidatePostSize
:27
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance
:110
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Http\Middleware\TrustProxies
:57
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
22 vendor frames collapsed

App\Http\Middleware\FixStatusCode
:30
handle
 
Illuminate\Pipeline\Pipeline
:183
Illuminate\Pipeline\{closure}
 
Illuminate\Pipeline\Pipeline
:119
then
 
Illuminate\Foundation\Http\Kernel
:175
sendRequestThroughRouter
 
Illuminate\Foundation\Http\Kernel
:144
handle
4 vendor frames collapsed

public/index.php
:73
vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php :503
vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php :483

### Steps to reproduce the issue

Go to "Maintenance" and click on fix tree.

### Diagnostics [REQUIRED]

 Diagnostics
    -----------
        Warning: Dropbox import not working. dropbox_key is empty.
        Warning: zend.assertions is disabled although Lychee is in debug mode. For easier debugging code generation for assertions should be enabled.
        Error: /v***************it (and subdirectories) are not executable, check the permissions
    
    System Information
    ------------------
        Lychee Version (git):                    master (656ffae) -- Up to date (-2.5009122647237E-6 years ago).
        DB Version:                              5.5.1
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               true
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.3.6
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                Europe/Berlin
        Max uploaded file size:                  500M
        Max post size:                           500M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      200
        MySQL Version:                           10.11.8-MariaDB-0ubuntu0.24.04.1-log
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1692
        GD Version:                              2.3.3
        Number of foreign key:                   12 found.
    
    Config Information
    ------------------
        version:                                 050501
        check_for_updates:                       1
        sorting_photos_col:                      taken_at
        sorting_photos_order:                    DESC
        sorting_albums_col:                      max_taken_at
        sorting_albums_order:                    DESC
        imagick:                                 1
        skip_duplicates:                         0
        small_max_width:                         0
        small_max_height:                        360
        medium_max_width:                        3840
        medium_max_height:                       2160
        lang:                                    en
        image_overlay_type:                      exif
        default_license:                         reserved
        compression_quality:                     90
        grants_full_photo_access:                1
        delete_imported:                         0
        mod_frame_enabled:                       1
        mod_frame_refresh:                       30
        thumb_2x:                                1
        small_2x:                                1
        medium_2x:                               1
        landing_page_enable:                     1
        force_migration_in_production:           1
        search_public:                           1
        hide_version_number:                     0
        SL_enable:                               0
        SL_for_admin:                            0
        recent_age:                              1
        grants_download:                         1
        photos_wraparound:                       1
        map_display:                             1
        zip64:                                   1
        map_display_public:                      1
        map_provider:                            RRZE
        force_32bit_ids:                         0
        map_include_subalbums:                   1
        update_check_every_days:                 3
        has_exiftool:                            1
        share_button_visible:                    1
        import_via_symlink:                      0
        has_ffmpeg:                              1
        location_decoding:                       1
        location_decoding_timeout:               30
        location_show:                           1
        location_show_public:                    1
        rss_enable:                              0
        rss_recent_days:                         7
        rss_max_items:                           100
        prefer_available_xmp_metadata:           1
        editor_enabled:                          1
        lossless_optimization:                   0
        swipe_tolerance_x:                       150
        swipe_tolerance_y:                       250
        local_takestamp_video_formats:           .avi|.mov
        log_max_num_line:                        1000
        unlock_password_photos_with_url_param:   0
        nsfw_visible:                            1
        nsfw_blur:                               1
        nsfw_warning:                            1
        nsfw_warning_admin:                      1
        nsfw_banner_override:                    
        map_display_direction:                   1
        album_subtitle_type:                     oldstyle
        upload_processing_limit:                 4
        new_photos_notification:                 0
        legacy_id_redirection:                   1
        zip_deflate_level:                       6
        default_album_protection:                1
        allow_username_change:                   1
        album_decoration:                        layers
        album_decoration_orientation:            row
        auto_fix_orientation:                    1
        use_job_queues:                          0
        random_album_id:                         starred
        use_last_modified_date_when_no_exif_date: 0
        layout:                                  justified
        date_format_photo_thumb:                 M j, Y, g:i:s A e
        date_format_photo_overlay:               M j, Y, g:i:s A e
        date_format_sidebar_uploaded:            M j, Y, g:i:s A e
        date_format_sidebar_taken_at:            M j, Y, g:i:s A e
        date_format_hero_min_max:                F Y
        date_format_hero_created_at:             M j, Y, g:i:s A T
        date_format_album_thumb:                 M Y
        upload_chunk_size:                       0
        nsfw_banner_blur_backdrop:               0
        search_pagination_limit:                 1000
        search_minimum_length_required:          4
        photo_layout_justified_row_height:       320
        photo_layout_masonry_column_width:       300
        photo_layout_grid_column_width:          250
        photo_layout_square_column_width:        200
        photo_layout_gap:                        12
        display_thumb_album_overlay:             always
        display_thumb_photo_overlay:             hover
        use_album_compact_header:                0
        default_album_thumb_aspect_ratio:        1/1
        login_button_position:                   left
        back_button_enabled:                     0
        back_button_text:                        Return to Home
        current_job_processing_visible:          1
        SA_random_thumbs:                        0
        job_processing_queue_visible:            1
        login_required:                          0
        enable_unsorted:                         1
        enable_starred:                          1
        enable_recent:                           1
        enable_on_this_day:                      1
        login_required_root_only:                1
    

### Browser & System [REQUIRED]

OS Ubuntu 24.04, client Windows10, Browser google chrome (latest).

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
Thank you. I will try to have a look.

**Reply by ildyria** [reply, marked as answer]
Disabled for now,
Fixing that one is trickier than expected and will need some UI work...
there is a fix tree function on nested set, I tried it locally and it pretty much flattened my set up, not enabling that for the other users.
